Source/PubnubChatSDK/Private/PubnubChat.cpp

```diff
@@ -20,8 +20,17 @@ UPubnubChat* UPubnubChat::Create(Pubnub::Chat Chat)
 	return NewChat;
 }
 
+void UPubnubChat::DestroyChat()
+{
+	delete InternalChat;
+	InternalChat = nullptr;
+	OnChatDestroyed.Broadcast();
```

Comment:
How the Broadcast works?

Reply:
This is something like "call" a delegate. Subsystem is listening for that, so it will know to clear Chat variable when chat is destroyed. Otherwise Subsystem will keep Chat as shared pointer for ever.